packages/app/src/runs/RunsErrorRenderer.vue

```diff
@@ -126,13 +126,27 @@ mutation RunsErrorRenderer_RequestAccess( $projectId: String! ) {
 }
 `
 
+const hasRequestedAccess = ref(false)
+
+onMounted(() => {
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectUnauthorized') {
+    hasRequestedAccess.value = props.gql.currentProject.cloudProject.hasRequestedAccess ?? false
+  }
+})
+
 const requestAccessMutation = useMutation(RunsErrorRenderer_RequestAccessDocument)
 
-function requestAccess () {
+async function requestAccess () {
   const projectId = props.gql.currentProject?.projectId
 
   if (projectId) {
-    requestAccessMutation.executeMutation({ projectId })
+    const result = await requestAccessMutation.executeMutation({ projectId })
```

Comment:
@mike-plummer I think you might have worked on this recently - any thoughts here?

Reply:
I restructured this a bit and used it as a pattern for `SpecHeaderCloudDataTooltip.vue` - my suspicion is that Muaz added the component-level state because the GQL state update wasn't circling back and causing the component to update. I don't know enough about the GQL stack yet to say whether that was expected behavior from the mutation, or whether this component needed some sort of subscription to be updated when that data updated. I agree that ideally this manual state management *shouldn't* be necessary - hopefully someone who knows more about our GQL setup can point me in the right direction.

Reply:
> Based on mutation result, GrpahQL cache (urql client) updates

Isn't this the kind of behavior we'd need a subscription to get now? That (since the removal of `liveMutation`) we're no longer syncing with the client?

I might be conflating two different situations though.

Reply:
> Isn't this the kind of behavior we'd need a subscription to get now?

No, in most cases this should happens automatically based on the requested fields in the mutation response. 

Looking at this again, I'm pretty sure the issue here is that because there isn't an identifier tying this to the locally resolved project, without what we're doing here, we need to add some updater logic to the `graphcacheExchange` to keep this in-sync correctly in the cache.

Reply:
Is this going to be a problem or can we dismiss this as "won't do"? Otherwise, do we need to file an issue (likely with something more detailed than "add a custom updater" since many people might not know what action is needed. What do you think?